compiler/src/iree/compiler/GlobalOptimization/GeneralizeLinalgNamedOps.cpp

```diff
@@ -31,32 +31,50 @@ struct GeneralizeLinalgNamedOpsPass
 };
 } // namespace
 
-/// Returns true of `linalgOp` is a Conv2DNchwFchwOp or Conv2DNhwcHwcfOp with
-/// all strides equal to 1 and with a kernel height and width of 1
+/// Returns true if `linalgOp` can be simplified to a basic GEMM.
 static bool isConvFoldableToContraction(linalg::LinalgOp linalgOp) {
-  auto NCHWOp = dyn_cast<linalg::Conv2DNchwFchwOp>(linalgOp.getOperation());
-  auto NHWCOp = dyn_cast<linalg::Conv2DNhwcHwcfOp>(linalgOp.getOperation());
+  auto convDimsOrFailure = linalg::inferConvolutionDims(linalgOp);
+  if (failed(convDimsOrFailure)) {
+    return false;
+  }
+  auto &convDims = *convDimsOrFailure;
 
-  if (!NCHWOp && !NHWCOp)
+  if (!llvm::all_of(convDims.strides,
+                    [](int64_t element) { return element == 1; })) {
     return false;
+  }
 
-  DenseIntElementsAttr strides =
-      NCHWOp ? NCHWOp.getStrides() : NHWCOp.getStrides();
-  if (!llvm::all_of(
-          strides, [](APInt element) { return element.getSExtValue() == 1; })) {
+  if (!llvm::all_of(convDims.dilations,
+                    [](int64_t element) { return element == 1; })) {
     return false;
   }
 
-  auto filterShapeType = llvm::dyn_cast<RankedTensorType>(
-      linalgOp.getDpsInputOperand(1)->get().getType());
-  if (!filterShapeType)
+  if (!convDims.depth.empty()) {
+    return false;
+  }
+
+  if (convDims.outputChannel.empty() || convDims.inputChannel.empty()) {
```

Comment:
nit: Add a comment here saying something like "Dont generalize pooling operations. For pooling ops, the input/output channel
  // size will be categorized as the additional batch dimension."